flycheck.el

```diff
@@ -5239,6 +5240,53 @@ This function determines the directory by looking at
     ;; If the buffer has no file name, fall back to its default directory
     default-directory))
 
+(defun flycheck-can-run-clang-check-p ()
+  "Check if clang-check can be run.
+
+For clang-check to run we need both a compile_commands.json database
+and for the buffer to be saved"
+  (and (flycheck-buffer-saved-p)
+       (eq 0 (call-process "clang-check" nil nil nil "dummy_file.c"))))
+
+(flycheck-define-checker c/c++-clang-check
+  "A C/C++ syntax checker using Clang libtooling checker.
+
+See URL `http://clang.llvm.org/docs/ClangCheck.html'.
+
+This checker works best if you create a compilation database to hold
+all the compile options for each file. See the URL
+`http://clang.llvm.org/docs/JSONCompilationDatabase.html' for details.
+"
+  :command ("clang-check"
+            "--extra-arg=-Wno-unknown-warning-option" ; silence GCC options
+            "--extra-arg=-Wno-null-character"         ; silence null
+            "--extra-arg=-fno-color-diagnostics"      ; Do not include color codes in output
+            "--extra-arg=-fno-caret-diagnostics"      ; Do not visually indicate the source
+            "--extra-arg=-fno-diagnostics-show-option" ; Do not show the corresponding
+            source-original)
+  :error-patterns
+  ((error line-start
+          (message "In file included from") " " (file-name) ":" line ":"
+          line-end)
+   (info line-start (file-name) ":" line ":" column
+         ": note: " (optional (message)) line-end)
+   (warning line-start (file-name) ":" line ":" column
+            ": warning: " (optional (message)) line-end)
+   (error line-start (file-name) ":" line ":" column
+          ": " (or "fatal error" "error") ": " (optional (message)) line-end))
+  :error-filter
+  (lambda (errors)
+    (let ((errors (flycheck-sanitize-errors errors)))
+      (dolist (err errors)
+        ;; Clang will output empty messages for #error/#warning pragmas without
+        ;; messages.  We fill these empty errors with a dummy message to get
+        ;; them past our error filtering
+        (setf (flycheck-error-message err)
+              (or (flycheck-error-message err) "no message")))
+      (flycheck-fold-include-levels errors "In file included from")))
+  :modes (c-mode c++-mode)
+  :predicate flycheck-can-run-clang-check-p)
```

Comment:
Turn this into a `lambda` please.  There's no use in named functions for predicates that are only used at one place.